docs: Update cli docs (#880)
apify/apify-cli: docs/troubleshooting.md

title: Troubleshooting
---

### Problems with installation

If you receive a permission error, read npm's [official guide](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally) on installing packages globally.

Alternatively, you can use [Node Version Manager (nvm)](https://github.com/nvm-sh/nvm) and install Apify CLI only into a selected user-level Node version without requiring root privileges:

```bash showLineNumbers
nvm install 18
nvm use 18
npm -g install apify-cli

> ```

Author: Changed.

title: Troubleshooting
---

> ### Problems with installation

Reviewer: ```suggestion
## Problems with installation
```

If you receive a permission error, read npm's [official guide](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally) on installing packages globally.

> Alternatively, use a version manager to install Node 20+. We recommend [`fnm` (Fast Node Manager)](https://github.com/Schniz/fnm) or [Volta](https://volta.sh):

Reviewer 2: So you can install node via apt on linux, and that installs npm in a way where global dependencies are installed under root. Its why we wanna recommend people install node solely through fnm/volta/god forbid nvm

> Alternatively, use a version manager to install Node 20+. We recommend [`fnm` (Fast Node Manager)](https://github.com/Schniz/fnm) or [Volta](https://volta.sh):

Reviewer: I mean, I understood it, I was more pointing out that we use `Alternatively` when other option is not spelled out there. I think if we just forego `alternatively` it makes better sense.

Maybe something like this:

```suggestion
A better approach is to use a Node.js version manager to install Node.js 20+. It prevents permission issues from happening in the first place. We recommend:

- [fnm (Fast Node Manager)](https://github.com/Schniz/fnm)
- [Volta](https://volta.sh/).
```

> #### Using fnm (recommended)

Reviewer: ```suggestion
### Using fnm (recommended)
```

```bash showLineNumbers
curl -fsSL https://fnm.vercel.app/install | bash
# Install and use Node 20
fnm install 20
fnm use 20
npm install -g apify-cli
```

> ### Using Volta (recommended)

Reviewer 2: I wouldn't even document how to install fnm, thats not on us to do! Just say "we recommend you install node using fnm/volta", and install with `npm install -g apify-cli`

> ### Using Volta (recommended)

Reviewer 2: Regardless of any of that, node 22 please!!

> ### Using Volta (recommended)

Reviewer: we should in the future make it more automated or at least make it so we need to change such things in one place (but this is a discussion for the future :D)

> #### Using Volta (recommended)

Reviewer: ```suggestion
### Using Volta (recommended)
```

```bash showLineNumbers
curl https://get.volta.sh | bash
# Install Node 20 and use it globally
volta install node@20
npm install -g apify-cli
```

:::note Last resort: nvm

If you prefer `nvm`, you can use it as well.

```bash showLineNumbers
nvm install 20
nvm use 20
npm -g install apify-cli
```

:::

> ### Migrations

Reviewer: ```suggestion
## Migrations
```

You can find the differences and migration info in [migration guidelines](https://github.com/apify/apify-cli/blob/master/MIGRATIONS.md).

> ### Help command

Reviewer: ```suggestion
## Help command
```
